### PR TITLE
Document how to report security issues

### PR DESCRIPTION
**Pull request by raboof**
### Description

Document how to report security issues on the security overview page, so we can link this page from the homepage. That should make all the other important security information easier to find as well.

Unfortunately I failed to build the site locally to test the change.

<hr>

This PR has:

- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [x] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

**Review by paul-rogers — APPROVED** [review]
LGTM. Nicely worded. Would recommend one of our Doc colleagues to give a quick review: @techdocsmith, @vtlim or @317brian.

The build failed, but it is very likely flaky. Restarted.

File: docs/operations/security-overview.md [hunk 1]
```diff
@@ -272,3 +272,19 @@ Cluster to deep storage:
 Cluster to client:
 1. Druid authenticates with the client based on the configured authenticator.
 2. Druid only performs actions when an authorizer grants permission. The default configuration is `allowAll authorizer`.
+
+## Reporting security issues
+
+The Apache Druid team takes security very seriously. If you have found a potential security issue in Druid, such as a way to bypass the security mechanisms described above, we encourage you to report this problem at [security@apache.org](mailto:security@apache.org). This is a private mailing list. Please send one plain-text email for each vulnerability you are reporting.
```
Comment (vtlim) [inline docs/operations/security-overview.md:278]:
```suggestion
The Apache Druid team takes security very seriously. If you find a potential security issue in Druid, such as a way to bypass the security mechanisms described earlier, please report this problem to [security@apache.org](mailto:security@apache.org). This is a private mailing list. Please send one plain text email for each vulnerability you are reporting.
```

File: docs/operations/security-overview.md [hunk 2]
```diff
@@ -272,3 +272,19 @@ Cluster to deep storage:
...
+
+### Vulnerability handling
+
+A summary of the vulnerability handling process is:
+
+* The reporter reports the vulnerability privately to [security@apache.org](mailto:security@apache.org)
+* You will receive a response that we have received the report and will investigate the issue.
```
Comment (vtlim) [inline docs/operations/security-overview.md:285]:
```suggestion
* The reporter receives a response that the Druid team has received the report and will investigate the issue.
```

File: docs/operations/security-overview.md [hunk 3]
```diff
@@ -272,3 +272,19 @@ Cluster to deep storage:
...
+* The Druid project security team works privately with the reporter to resolve the vulnerability.
+* The Druid project creates a new release of the package the vulnerabilty affects to deliver its fix.
+* The Druid project publicly announces the vulnerability and describes how to apply the fix.
```
Comment (vtlim) [inline docs/operations/security-overview.md:288]:
```suggestion
* The Druid team publicly announces the vulnerability and describes how to apply the fix.
```

File: docs/operations/security-overview.md [hunk 4]
```diff
@@ -272,3 +272,19 @@ Cluster to deep storage:
...
+The Apache Druid team takes security very seriously. If you have found a potential security issue in Druid, such as a way to bypass the security mechanisms described above, we encourage you to report this problem at [security@apache.org](mailto:security@apache.org). This is a private mailing list. Please send one plain-text email for each vulnerability you are reporting.
+
+### Vulnerability handling
+
+A summary of the vulnerability handling process is:
+
+* The reporter reports the vulnerability privately to [security@apache.org](mailto:security@apache.org)
+* You will receive a response that we have received the report and will investigate the issue.
+* The Druid project security team works privately with the reporter to resolve the vulnerability.
+* The Druid project creates a new release of the package the vulnerabilty affects to deliver its fix.
```
Comment (vtlim) [inline docs/operations/security-overview.md:287]:
```suggestion
* The Druid team delivers the fix by creating a new release of the package that the vulnerability affects.
```

File: docs/operations/security-overview.md [hunk 5]
```diff
@@ -272,3 +272,19 @@ Cluster to deep storage:
...
 1. Druid authenticates with the client based on the configured authenticator.
 2. Druid only performs actions when an authorizer grants permission. The default configuration is `allowAll authorizer`.
+
+## Reporting security issues
+
+The Apache Druid team takes security very seriously. If you have found a potential security issue in Druid, such as a way to bypass the security mechanisms described above, we encourage you to report this problem at [security@apache.org](mailto:security@apache.org). This is a private mailing list. Please send one plain-text email for each vulnerability you are reporting.
+
+### Vulnerability handling
+
+A summary of the vulnerability handling process is:
```
Comment (vtlim) [inline docs/operations/security-overview.md:282]:
```suggestion
The following list summarizes the vulnerability handling process:
```

**Review by vtlim — COMMENTED** [review]
Left some suggestions for improving clarity and style.

**Comment by raboof** [comment]
> Left some suggestions for improving clarity and style.

Great, thanks, applied!

**Review by vtlim — APPROVED** [review]
LGTM, thank you!

**Comment by paul-rogers** [comment]
Our docs checker does not like `mailto` links. Anyone know how to fix this?

File: docs/operations/security-overview.md [hunk 6]
```diff
@@ -272,3 +272,19 @@ Cluster to deep storage:
 Cluster to client:
 1. Druid authenticates with the client based on the configured authenticator.
 2. Druid only performs actions when an authorizer grants permission. The default configuration is `allowAll authorizer`.
+
+## Reporting security issues
+
+The Apache Druid team takes security very seriously. If you find a potential security issue in Druid, such as a way to bypass the security mechanisms described earlier, please report this problem to [security@apache.org](mailto:security@apache.org). This is a private mailing list. Please send one plain text email for each vulnerability you are reporting.
```
Comment (abhishekagarwal87) [inline docs/operations/security-overview.md:278]: can you remove the link and just put email id as it is? docs checker seem to be failing here.

**Review by abhishekagarwal87 — APPROVED** [review]

**Pull request merged**